src/org/opensolaris/opengrok/util/Executor.java

```diff
@@ -217,12 +232,21 @@ public void run() {
             handler.processStream(process.getInputStream());
 
             ret = process.waitFor();
+            
             LOGGER.log(Level.FINE,
                 "Finished command {0} in directory {1}",
                 new Object[] {cmd_str,dir_str});
-            process = null;
+
+            // Wait for the stderr read-out thread to finish the processing and
+            // only after that read the data.
             thread.join();
             stderr = err.getBytes();
+            
+            // Close the pipes to avoid file descriptors from being drained
+            // quickly. Then set the process object to null in the high hopes
+            // that the garbage collector will finish rest of the cleanup soon.
+            closeStreams(process);
```

Comment:
Maybe I am missing something but why are you closing the process here and then you are doing the same in `finally`? I think this is basically redundant...

Reply:
true, moved the close to the finally block. I never liked setting objects to null anyway.